lib/src/repo_path.rs

```diff
@@ -217,6 +227,15 @@ impl Debug for RepoPathBuf {
     }
 }
 
+/// The `value` is not a valid repo path because it contains empty path
+/// component. For example, `"/"`, `"/foo"`, `"foo/"`, `"foo//bar"` are all
+/// invalid.
+#[derive(Clone, Debug, Eq, Error, PartialEq)]
+#[error(r#"Invalid repo path input "{value}""#)]
+pub struct InvalidNewRepoPathError {
+    value: String,
+}
```

Comment:
nit: It seems inconsistent that we add new error type for `RepoPath::from_internal_string()`, but reuse the non-syntactic error type for `RepoPathComponent::new()`.

I don't have strong feeling whether we should add/reuse the error types, but it might be easier to use this `InvalidNewRepoPathError` type for malformed inputs of both `RepoPath` and `RepoPathComponent`.

Reply:
Makes sense. Switched to `InvalidNewRepoPathError`